### ISSUE TITLE
Tests fail when executed from umbrella app in global mode.

### ISSUE THREAD

**Issue by alex-sysoev** [plataformatec/mox#31]
We have a function that launches a Task and calls two functions from different behaviours.
We configured two mocks and called expect in each of them but it fails and says that the second expectation was not defined.
We also tried configuring one mock with both behaviours (using the version in master branch), but it fails with the same reason.

The strange part is that if there's only one test, and it is executed from inside the particular app's directory, the test passes. Either if you run the tests from the umbrella app's directory or if you add a second test, it fails.

This is the error:
```elixir
15:27:24.526 [error] Task #PID<0.189.0> started from #PID<0.188.0> terminating
** (Mox.UnexpectedCallError) no expectation defined for App1.SecondMock.bye/0 in process #PID<0.189.0>
    (mox) lib/mox.ex:466: Mox.__dispatch__/4
    (elixir) lib/task/supervised.ex:88: Task.Supervised.do_apply/2
    (stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
Function: #Function<0.6366682/0 in App1.test_case/0>
    Args: []
```

[Here is a link to a sample project that reproduces the failure](https://github.com/alex-sysoev/mox-umbrella).

**Comment by josevalim** [comment]
It is a race condition. Because you don't wait for the task to finish, if you add `setup :verify_on_exit!`, you will see that your expectations are not being invoked because the task did not run yet.

When the test finishes, it will "reclaim" all of the expectations that you have defined. So by the time the task actually runs, the expectations are no longer there, and the test fails.

You only see it in an umbrella because running the tests of the second app give you enough time for the tasks to finish. When you run a single app, the whole suite finishes before the tasks.

TL;DR - use `setup :verify_on_exit!`.

**Comment by josevalim** [comment]
Btw, thanks for the mechanism to reproduce the issue. It makes everything much easier! :)

**Comment by alex-sysoev** [comment]
Thank you for great explanation, Jose !:)
